packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -444,12 +461,42 @@ export abstract class QueryBuilderState implements CommandRegistrar {
     return PrimitiveType.STRING;
   }
 
-  initializeWithQuery(query: RawLambda): void {
-    this.rebuildWithQuery(query);
+  initializeWithQuery(
+    query: RawLambda,
+    defaultParameterValues?: Map<string, ValueSpecification>,
+  ): void {
+    this.rebuildWithQuery(query, {
+      defaultParameterValues,
+    });
     this.resetQueryResult();
     this.changeDetectionState.initialize(query);
   }
 
+  buildInitialParameterValues(
+    preserveParameterValues: boolean | undefined,
+    defaultParameterValues: Map<string, ValueSpecification> | undefined,
+  ): Map<string, QueryBuilderParameterValue> | undefined {
+    const res = new Map<string, QueryBuilderParameterValue>();
+    if (defaultParameterValues?.size) {
+      Array.from(defaultParameterValues.entries()).forEach(([k, v]) => {
+        res.set(k, {
+          variable: k,
+          value: v,
+        });
+      });
+      return res;
+    } else if (preserveParameterValues) {
+      this.parametersState.parameterStates.forEach((ps) => {
+        res.set(ps.parameter.name, {
+          variable: ps.parameter,
+          value: ps.value,
+        });
+      });
+      return res;
+    }
+    return undefined;
```

Comment:
Also, I would expect to logic here to be merging existing one with default ones where the former override, for example I'm running my query A

- A has default param: x = 1, y = 2
- We run A set x = 2, so we have x = 2, y = 2

Reply:
no i dont think we should merge. 
The expectation is really one or the other. The initial load of the query will have either defaultParameters or nothing. 
Each rebuild after has a defined state build and will use that.